lib/validate.js

```diff
@@ -18,6 +18,9 @@ module.exports = {
         throw new this.serverless.classes
           .Error('The webpack plugin could not find the configuration file at: ' + webpackConfigFilePath);
       }
+      if (this.options.function) {
+        process.env.serverlessFunction = this.options.function;
```

Comment:
Is there any reason why an additional environment variable is set here? I would not set anything in the process' environment as it is global. Better would be to set it as member of the plugin object with `this.XXX = ...` in case it is needed by the plugin itself.
If it is meant to be used in the vars of serverless.yml, you can reference it via options as `${opt:function}` there, even with specifying a default in case it is not set (`${opt:function, 'none'}`).